internal/test/e2e/version_compatibility_test.go

```diff
@@ -1283,3 +1283,56 @@ func TestVersionCompatibility_Zipkin310OrNewer(t *testing.T) {
 		return err
 	})
 }
+
+func TestVersionCompatibility_KeysEntities(t *testing.T) {
+	cleanup := run.Koko(t)
+	defer cleanup()
+
+	dpCleanup := run.KongDP(kong.GetKongConfForShared())
+	defer dpCleanup()
+	require.NoError(t, util.WaitForKong(t))
+	require.NoError(t, util.WaitForKongAdminAPI(t))
+
+	admin := httpexpect.WithConfig(httpexpect.Config{
+		BaseURL:  "http://localhost:3000",
+		Reporter: httpexpect.NewRequireReporter(t),
+		Printers: []httpexpect.Printer{
+			httpexpect.NewCompactPrinter(t),
+		},
+	})
+
+	admin.POST("/v1/keys").WithJSON(&v1.Key{
+		Name: "second",
+		Jwk:  &v1.JwkKey{},
+	}).Expect().Status(http.StatusCreated)
+
+	t.Run("pre 3.0", func(t *testing.T) {
+		kongClient.RunWhenKong(t, "< 3.0.0")
+
+		util.WaitFunc(t, func() error {
+			return util.EnsureConfig(&v1.TestingConfig{})
+		})
+	})
+
+	t.Run("pre 3.1", func(t *testing.T) {
+		kongClient.RunWhenKong(t, ">= 3.0.0 < 3.1.0")
```

Comment:
Maybe I'm missing something, but couldn't this test and the `pre 3.0` one be converged in a single `pre 3.1`? How is `pre 3.0` different than `pre 3.1`?

Reply:
yes, `pre 3.0` is redundant; i added it to see how the current "master" image identifies as.  it seems it still doesn't say 3.1, so the test isn't too useful yet (unless there's some 3.1 image somewhere).